src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionsStateBase.cs

```diff
@@ -15,190 +15,13 @@
-    {
-        return GetSubscriptionStateKey(context, database, subscriptionId, currentChangeVector, SubscriptionType.Revision, out key);
-    }
-
-    public static unsafe ByteStringContext<ByteStringMemoryCache>.InternalScope GetDatabaseAndSubscriptionPrefix(ClusterOperationContext context, string database, long subscriptionId, out ByteString prefix)
-    {
-        using var _ = Slice.From(context.Allocator, database.ToLowerInvariant(), out var dbName);
-        var rc = context.Allocator.Allocate(dbName.Size + sizeof(byte) + sizeof(long) + sizeof(byte), out prefix);
-
-        dbName.CopyTo(prefix.Ptr);
-        var position = dbName.Size;
-
-        *(prefix.Ptr + position) = SpecialChars.RecordSeparator;
-        position++;
-
-        *(long*)(prefix.Ptr + position) = subscriptionId;
-        position += sizeof(long);
-
-        *(prefix.Ptr + position) = SpecialChars.RecordSeparator;
-
-        return rc;
-    }
-
-    public static unsafe ByteStringContext<ByteStringMemoryCache>.InternalScope GetSubscriptionStateKey(ClusterOperationContext context, string database, long subscriptionId, string pk, SubscriptionType type, out ByteString key)
-    {
-        switch (type)
-        {
-            case SubscriptionType.Document:
-                pk = pk.ToLowerInvariant();
-                break;
-            case SubscriptionType.Revision:
-                break;
-            default:
-                throw new ArgumentOutOfRangeException(nameof(type), type, null);
-        }
-
-        using var _ = Slice.From(context.Allocator, database.ToLowerInvariant(), out var dbName);
-        using var __ = Slice.From(context.Allocator, pk, out var pkSlice);
-        var rc = context.Allocator.Allocate(dbName.Size + sizeof(byte) + sizeof(long) + sizeof(byte) + sizeof(byte) + sizeof(byte) + pkSlice.Size, out key);
-
-        PopulatePrefix(subscriptionId, type, ref key, ref dbName, out int position);
-
-        pkSlice.CopyTo(key.Ptr + position);
-        return rc;
-    }
-        
-    private static unsafe void PopulatePrefix(long subscriptionId, SubscriptionType type, ref ByteString prefix, ref Slice dbName, out int position)
-    {
-        dbName.CopyTo(prefix.Ptr);
-        position = dbName.Size;
-
-        *(prefix.Ptr + position) = SpecialChars.RecordSeparator;
-        position++;
-
-        *(long*)(prefix.Ptr + position) = subscriptionId;
-        position += sizeof(long);
-
-        *(prefix.Ptr + position) = SpecialChars.RecordSeparator;
-        position++;
-
-        *(prefix.Ptr + position) = (byte)type;
-        position++;
-
-        *(prefix.Ptr + position) = SpecialChars.RecordSeparator;
-        position++;
-    }
-}
-
-public abstract class SubscriptionConnectionsStateBase<TSubscriptionConnection, TIncludesCommand> : SubscriptionConnectionsStateBase, IDisposable
+public abstract class SubscriptionConnectionsStateBase<TSubscriptionConnection, TIncludesCommand> : AbstractSubscriptionConnectionsState
```

Comment:
I would like to rename `AbstractSubscriptionConnectionsState` -> `SubscriptionConnectionsStateBase`

Reply:
note, that this isn't something that should block the merge :)

Reply:
yes